src/game/gameflow.c

```diff
@@ -1299,3 +1299,114 @@ GameFlow_InterpretSequence(int32_t level_num, GAMEFLOW_LEVEL_TYPE level_type)
 
     return ret;
 }
+
+GAMEFLOW_OPTION
+GameFlow_StorySoFar(
+    int32_t level_num, GAMEFLOW_LEVEL_TYPE level_type, int32_t savegame_level)
+{
+    LOG_INFO("%d", level_num);
+
+    GAMEFLOW_SEQUENCE *seq = g_GameFlow.levels[level_num].sequence;
+    GAMEFLOW_OPTION ret = GF_EXIT_TO_TITLE;
+    while (seq->type != GFS_END) {
+        LOG_INFO("seq %d %d", seq->type, seq->data);
+
+        switch (seq->type) {
+        case GFS_LOOP_GAME:
+        case GFS_STOP_GAME:
+        case GFS_LEVEL_STATS:
+        case GFS_TOTAL_STATS:
+        case GFS_DISPLAY_PICTURE:
+        case GFS_GIVE_ITEM:
+        case GFS_REMOVE_GUNS:
+        case GFS_REMOVE_SCIONS:
+        case GFS_FIX_PYRAMID_SECRET_TRIGGER:
+            seq++;
+            continue;
+
+        case GFS_START_GAME:
+            if (level_num == savegame_level) {
+                return GF_EXIT_TO_TITLE;
+            }
+            seq++;
+            continue;
```

Comment:
What's this check about? Why not just skip it?

Reply:
I added this check:
```
            if (level_num == savegame_level) {
                return GF_EXIT_TO_TITLE;
            }
```
because levels like the Great Pyramid were playing the ending FMV when doing the story so far. This check makes it so the function reads the current level but stops at the point of `start_game`. So the story so far will play any cutscenes/FMVs that occur before the level, but will not play anything that comes after `start_game`.